### PR TITLE
Feature/3528 usermenu examples

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3528

#### Changes proposed in this Pull Request:
- Added examples page for User Menu

#### Screenshots:
Screenshot 2022-09-26 at 5 58 22 PM

#### To Test:
- yarn start:reactdev

### CODE REVIEW

**Comment by emclaug2** [comment]
Most of my comments are around what is highlighted. 

For "User Menu with non-text Avatars", there's too many lines highlighted.  Instead, highlight lines "3, 16-20" instead (just the avatar).

For "User Menu with a Menu Header", highlight the `menuTitle` and `menuSubtitle` prop.

For "User Menu with a Custom Header", highlight the custom header.

For "User Menu with Placement Options", highlight the `MenuProps`.  Also is there a link that can be provided here for more context?  What are `MenuProps`? Are they from us, from MUI?  

For "Responsive User Menu", highlight the `useBottomSheetAt` prop.

File: docs/src/componentDocs/UserMenu/examples/UserMenuWithCustomHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Box } from '@mui/material';
+import { CodeBlock, CodeBlockActionButtonRow } from '../../../shared';
+import * as colors from '@brightlayer-ui/colors';
+import { UserMenuWithCustomHeaderExample } from './UserMenuWithCustomHeaderExample';
+
+const codeSnippet = `<UserMenu
+    onClick={(): void => setOpen(!open)}
+    avatar={<Avatar src={tRex} alt={'User Avatar'} />}
```
Comment (ektasawant) [inline docs/src/componentDocs/UserMenu/examples/UserMenuWithCustomHeader.tsx:9]: @bkarambe , can we add a string here? Right now it is showing file name.
Screenshot 2022-09-27 at 11 14 32 AM

**Review by jeffvg — APPROVED** [review]
looks good

**Comment by emclaug2** [comment]
image

Just these lines for the UserMenu Header since it's talking about the header portion.

**Comment by emclaug2** [comment]
Other than my previous comment everything else looks good to me!

**Pull request merged**